feat(sig-services): Create add or remove user requirement (#105)
open-services-group/community: sig-services/service-requirements/add-or-remove-user.md

---
title: Easy to add or remove user
persona: Service-Consumer (subscribed) with Service-Admin role (ACL)
phase: Service-operational
---

> This requirement states how should be handled subscribing and unsubscribing from the service. Users should be able to subscribe or unsubscribe easily (ideally by clicking one button).

Reviewer: ```suggestion
Users should be able to subscribe to and unsubscribe from a service easily. Both processes should be made as accessible to users as possible.

Subscription process should be as simple as registering via a single button while filling in the required minimal amount of user identification data. When subscribing to the service, users are not required to share more personally identifiable information than the mandatory identification or billing data. Subscription process is automated and user is approved and cleared to consume the service promptly. 
```

Author: Resolved in the latest commit.

> All linked accounts should be removed when unsubscribing from the service, and the user data should be deleted.

Reviewer: ```suggestion
A termination of a subscription is handled in similar fashion. The button is clearly visible and accessible to the user. It's also a simple process. Remember that hiding the unsubscribe button from users or making the process more complicated than necessary doesn't raise customer's satisfaction with the service.

When user unsubscribes from the service it is guaranteed that all linked accounts are removed from the system in that instant and user account is fully removed. All the user data is forever deleted from the service data store and it is ensured the same happens with the user's data passed to any third party service.
```

Author: Resolved in the latest commit.

## Requirements

> - After authentication, users should be able to subscribe or unsubscribe from a service by clicking one button.

Reviewer: ```suggestion
- User is able to subscribe to the service by clicking a single button. Authentication can be delegated to third party, but only mandatory identification and billing data is required from the user.
- User is able to unsubscribe from the service by clicking a single button (single confirmation layer is possible). The process is simple and the button is accessible and easy to find in the service settings.
```

Author: Resolved in the latest commit.

> - When a user unsubscribes from the service, all linked accounts should be removed from the service.

Reviewer: ```suggestion
- When a user unsubscribes from the service, all linked accounts are removed from the service.
```

Author: Resolved in the latest commit.

> - When a user unsubscribes from the service, all user data should be deleted.

Reviewer: ```suggestion
- When a user unsubscribes from the service, all user data is ensured to be deleted.
```

## Reasoning

> Users should be able to quickly onboard to the service and easily remove themselves from the service. This ensures that users can start using services immediately, and when they wish not to use the service, they don't have to worry about their data being stored after unsubscribing from the service.

Reviewer: ```suggestion
Users should be able to quickly onboard to the service and easily remove themselves from the service. This ensures that users can start using services immediately. And when they wish to stop using the service, they don't have to worry about their data being stored after unsubscribing from the service.
```

Author: Resolved in the latest commit.
